libcalico-go/lib/ipam/ipam_types.go

```diff
@@ -172,3 +173,22 @@ type BlockArgs struct {
 	// If specified, the attributes of reserved IPv6 addresses in this block.
 	HostReservedAttrIPv6s *HostReservedAttr
 }
+
+type ReleaseOptions struct {
```

Comment:
You could make this private and provide a constructor that requires the IP to be specified

Reply:
I had also considered going full-on options pattern here, but the exising ReleaseIPs function already takes mutliple IPs as its argument, and I decided I didn't have the appetite to refactor it to take only a single IP at this point in time. 

e.g.

```go
func ReleaseIP(ctx context.Context, ip string, opts ...ReleaseOption) {}